source3/modules/vfs_winmsa.c

```diff
@@ -251,6 +246,13 @@ static int winmsa_rename(struct vfs_handle_struct *handle,
 		goto out;
 	}
 
+	dst = talloc_size(ctx, PATH_MAX);
+	if (realpath(smb_fname_dst->base_name, dst) == NULL) {
```

Comment:
Wouldn't it be more efficient to make this call first and then based on a result just calculate parent dir?